### ISSUE TITLE
`pytests` without internet connection

### ISSUE THREAD

**Issue by LecrisUT** [scikit-build/scikit-build-core#246]
Are there any tags or other ways to filter out all pytests that require internet? For packaging, usually it is the default to disable internet connection.

Also how crucial are these tests and is it possible to mock them somehow?

**Comment by henryiii** [comment]
`—m “not isolated”`. Though I’m not totally sure that would disable the prepare step that does the download from PyPI - I think it does though. Fixtures shouldn’t run if never requested.

**Comment by LecrisUT** [comment]
Yeah, we had the `-m not isolated`, but it didn't work. [Here](https://download.copr.fedorainfracloud.org/results/lecris/scikit-build-core/fedora-rawhide-x86_64/05689486-python-scikit-build-core/builder-live.log.gz) is an example build where that failed.

**Comment by henryiii** [comment]
Oops, looks like that snuck in with #197, at https://github.com/scikit-build/scikit-build-core/blob/1e2b7cfccbedd2f29995819950ce1b04ceba54a5/tests/test_dynamic_metadata.py#L218
